lms/djangoapps/program_enrollments/models.py

```diff
@@ -55,7 +56,7 @@ class Meta(object):
     external_user_key = models.CharField(
         db_index=True,
         max_length=255,
-        null=True
+        default=DEFAULT_EXTERNAL_USER_KEY
```

Comment:
Not 100% sure how Django ORM constraints work.... do we need to make this non-null for the constraint to be enforcable?

Is it the case that if `unique_together('X', 'Y')`, with Y being nullable, the following would be allowed?
```
id | X     | Y
__________________
0  | a     | b
0  | NULL  | c
0  | NULL  | c
```

Reply:
After looking at the docs, it looks like `unique_together` does indeed mean "unique across non-null values", per the SQL philosophy that `NULL != NULL`.

I don't love sentinels but I guess it's necessary here.